### PR TITLE
Fixed Maven artifact resolver initialization in QuarkusProjectMojoBase

### PR DESCRIPTION
**Pull request by aloubyansky**
This fixes goals such as `quarkus:list-platforms` executed outside of project directory.

### CODE REVIEW

**Comment by geoand** [comment]
The test failures seems suspicious

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/MavenProjectBuildFile.java [hunk 1]
```diff
@@ -86,8 +86,8 @@ public static QuarkusProject getProject(Artifact projectPom, Model projectModel,
         final List<ArtifactCoords> importedPlatforms;
         final String quarkusVersion;
         if (projectPom == null) {
-            managedDeps = Collections.emptyList();
-            deps = () -> Collections.emptyList();
+            managedDeps = List.of();
+            deps = List::of;
```
Comment (gastaldi) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/MavenProjectBuildFile.java:90]: Out of curiosity, what is the difference between both?

**Review by gastaldi — APPROVED** [review]
Tested locally and it works! Good job!

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/MavenProjectBuildFile.java (hunk 1 above)
Comment (aloubyansky) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/MavenProjectBuildFile.java:90]: In this case, it's practically equivalent. I probably modified this class in some other way originally and then reverted but these lines stayed.

**Pull request merged**